File: test/matlab_comparison/signal_processing/anti_aliasing/anti_aliasing.m
fs = 2048;
fn = fs/2;

% Filter design
order = 200;
fb = 0.05*fn;
b = firls(order, [0 1/2-0.05 1/2+0.05 1] , [1, 1, 0, 0]);

[h, w] = freqz(b, 1);
freq = w/pi*fs;

% Visualization
figure
subplot(2, 1, 1)
plot(freq, abs(h))
subplot(2, 1, 2)
plot(freq, unwrap(angle(h)))

hf = h(freq <= fn);
freq = freq(freq <= fn);

% save test_anti_aliasing.mat hf freq

File: test/matlab_comparison/signal_processing/anti_aliasing/anti_aliasing.jl
using StructuralVibration, DSP
@usingany GLMakie, MAT

vars = matread("test/matlab_comparison/signal_processing/anti_aliasing/test_anti_aliasing.mat")

hf = vec(vars["hf"])

fs = 4096
fn = fs/2

# Filter design
order = 200     # Filter order
fb = 0.05*fn    # Filter bandwith = 2*fb
freq_filt = [(0., fn - fb) => 1]
filt_coeff = remez(order, freq_filt, Hz = fs, maxiter = 50)

# Filter visualization
z =  PolynomialRatio(filt_coeff, [1.0])
h, w = freqresp(z)
freq = w*fn/π

y = [transpose(hf); transpose(h)]

bode_plot(freq, y)

# Filtering
x = randn(fs)

y = filtfilt(filt_coeff, x)